src/App.tsx

```diff
@@ -1,24 +1,214 @@
+
+      return [];
+    });
+
+    setIsToggleActiveTodos(!isToggleActiveTodos);
+
+    Promise.all(promiseList).then(() => {
+      getTodos(user.id).then((todoList) => {
+        setTodos(todoList);
+        setIsShowFooter(Boolean(todoList.length));
+      }).catch(err => new Error(err.message))
+        .finally(() => setIsLoading(false));
+    });
+  };
+
+  const checkCompletedTodo = (arr: Todo[]) => {
+    isUncomplete = 0;
+    for (let i = 0; i < arr.length; i += 1) {
+      if (!arr[i].completed) {
+        isUncomplete += 1;
+      }
+    }
+  };
+
+  const deleteTodo = (id: number) => {
+    setIsLoading(true);
+
+    return deleteTodoOnServer(id)
+      .then(() => {
+        getTodos(user.id).then((todoList) => {
+          setTodos(todoList);
+          checkCompletedTodo(todoList);
+          setIsShowFooter(Boolean(todoList.length));
+        });
+      })
+      .catch(() => setError(ResponseError.ADD))
+      .finally(() => setIsLoading(false));
+  };
+
+  const deleteCompletedTodo = () => {
+    getTodos(user.id, 'completed=true')
+      .then((todoList) => {
+        return Promise.all(
+          todoList.map((todo: Todo) => deleteTodoOnServer(todo.id)),
+        );
+      })
+      .then(() => getTodos(user.id))
+      .then((todoList) => {
+        setTodos(todoList);
+        checkCompletedTodo(todoList);
+        setIsShowFooter(Boolean(todoList.length));
+      })
+      .catch((errorresp) => new Error(errorresp.message))
+      .finally(() => setIsLoading(false));
+  };
+
+  const updateTodo = (todoId: number, obj: Partial<Todo>) => {
+    setIsLoading(true);
+    updateTodoProp(todoId, obj)
+      .then(() => {
+        getTodos(user.id).then((todoList) => {
+          setTodos(todoList);
+          checkCompletedTodo(todoList);
+          setIsShowFooter(Boolean(todoList.length));
+        });
+      })
+      .catch(() => setError(ResponseError.UPDATE))
+      .finally(() => setIsLoading(false));
+  };
+
+  const displayTodos = (sortBy: Etodos) => {
+    switch (sortBy) {
+      case Etodos.ACTIVE:
+        return todos.filter((todo) => todo.completed === false);
+
+      case Etodos.COMPLETED:
+        return todos.filter((todo) => todo.completed === true);
+
+      default:
+        return [...todos];
+    }
+  };
+
+  const headerAddTodo = (str: string) => {
+    createTodo(str.trim(), user.id)
+      .then(() => {
+        getTodos(user.id).then((todoList) => {
+          setTodos(todoList);
+          checkCompletedTodo(todoList);
+          setIsShowFooter(Boolean(todoList.length));
+          setCreatingTodoTitle('');
+        });
+      })
+      .catch(() => setError(ResponseError.ADD))
+      .finally(() => setIsLoading(false));
+  };
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos(user.id).then((todosList) => {
+      checkCompletedTodo(todosList);
+      setTodos(todosList);
+      setIsLoading(false);
+    }).catch(err => new Error(err.message))
+      .finally(() => setIsLoading(false));
```

Comment:
You already have `setIsLoading(false);` in `finally` case, do it's redundant for `then`
```suggestion
    getTodos(user.id).then((todosList) => {
      checkCompletedTodo(todosList);
      setTodos(todosList);
    }).catch(err => new Error(err.message))
      .finally(() => setIsLoading(false));
```